plugin/src/main/java/io/jenkins/plugins/casc/yaml/YamlSource.java

```diff
@@ -1,86 +1,46 @@
 package io.jenkins.plugins.casc.yaml;
 
-import java.io.BufferedReader;
-import java.io.IOException;
 import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.io.Reader;
-import java.net.URI;
-import java.net.URL;
-import java.nio.file.Files;
 import java.nio.file.Path;
 import javax.servlet.http.HttpServletRequest;
 
-import static java.nio.charset.StandardCharsets.UTF_8;
-
 /**
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
 
-public class YamlSource<T> implements AutoCloseable {
-
-    public static final YamlReader<String> READ_FROM_URL = config -> {
-        final URL url = URI.create(config).toURL();
-        return new InputStreamReader(url.openStream(), UTF_8);
-    };
-
-    public static final YamlReader<Path> READ_FROM_PATH = Files::newBufferedReader;
-
-    public static final YamlReader<InputStream> READ_FROM_INPUTSTREAM = in -> new InputStreamReader(in, UTF_8);
-
-    public static final YamlReader<HttpServletRequest> READ_FROM_REQUEST = req -> {
-        // TODO get encoding from req.getContentType()
-        return new InputStreamReader(req.getInputStream(), UTF_8);
-    };
```

Comment:
These should never have been public 😢
See the wrong usage over here: https://github.com/jenkinsci/kubernetes-operator/blob/ee01c4c0285644af26148e6367678bd7da0d997d/pkg/configuration/user/casc/casc.go#L47-L58